File: app/src/main/java/org/mifos/mobilewallet/qr/QrContract.java
package org.mifos.mobilewallet.qr;

import android.graphics.Bitmap;

import org.mifos.mobilewallet.base.BasePresenter;
import org.mifos.mobilewallet.base.BaseView;

/**
 * Created by naman on 8/7/17.
 */

public interface QrContract {

    interface ShowQrView extends BaseView<ShowQrPresenter> {

        void showGeneratedQr(Bitmap bitmap);
    }

    interface ShowQrPresenter extends BasePresenter {

        void generateQr(String data);
    }
}

File: mifospay/src/main/java/org/mifospay/qr/QrContract.kt
package org.mifospay.qr

import android.graphics.Bitmap
import org.mifospay.base.BasePresenter
import org.mifospay.base.BaseView

/**
 * Created by naman on 8/7/17.
 */
interface QrContract {
    interface ShowQrView : BaseView<ShowQrPresenter?> {
        fun showGeneratedQr(bitmap: Bitmap?)
    }

    interface ShowQrPresenter : BasePresenter {
        fun generateQr(data: String?)
    }

    interface ReadQrView : BaseView<ReadQrPresenter?>
    interface ReadQrPresenter : BasePresenter
}